dashboard/app/controllers/api/v1/ml_models_controller.rb

```diff
@@ -4,10 +4,15 @@
 class Api::V1::MlModelsController < ApplicationController
   skip_before_action :verify_authenticity_token
 
-  # POST api/v1//ml_models/save
-  # Save a trained ML model to S3
+  # POST api/v1/ml_models/save
+  # Save a trained ML model to S3 and a reference to it in the database.
   def save
     model_id = generate_id
+    UserMlModel.create!(
+      user_id: current_user.id,
```

Comment:
if `current_user` is required, we should use CanCan to authorize this request. then, if there's no user, the request will return a 400 instead of trying to save to the database and 500ing

Reply:
this controller action also has a couple of distinct steps now (e.g., making sure we don't upload to s3 if UserMlModel creation fails), so we should probably cover it with unit tests if we haven't already